go/vt/vtgate/engine/update.go

```diff
@@ -64,7 +65,18 @@ func (upd *Update) GetKeyspaceName() string {
 // GetTableName specifies the table that this primitive routes to.
 func (upd *Update) GetTableName() string {
 	if upd.Table != nil {
-		return upd.Table.Name.String()
+		tableNameMap := map[string]any{}
```

Comment:
this is the same code as for delete. we should extract it into a helper method on DML